diff --git a/src/nautilus-pathbar.c b/src/nautilus-pathbar.c
index 2a7bdc4..542d12f 100644
--- a/src/nautilus-pathbar.c
+++ b/src/nautilus-pathbar.c
@@ -322,6 +322,16 @@ get_dir_name (ButtonData *button_data)
             return _("Starred");
         }
 
+        case RECENT_BUTTON:
+        {
+            return _("Recent");
+        }
+
+        case TRASH_BUTTON:
+        {
+            return _("Trash");
+        }
+
         default:
         {
             return button_data->dir_name;
