From f567696a004ebcaf45871c209015c29933d1c8b7 Mon Sep 17 00:00:00 2001
From: donghun <donghun@gooroom.kr>
Date: Wed, 16 Feb 2022 12:40:18 +0900
Subject: [PATCH] Added patch for change file extraction method in context menu

---
 src/nautilus-files-view.c | 88 ++++++++++++++++++++++++++++++++++++++-
 1 file changed, 87 insertions(+), 1 deletion(-)

diff --git a/src/nautilus-files-view.c b/src/nautilus-files-view.c
index 1c05680..be36ac5 100644
--- a/src/nautilus-files-view.c
+++ b/src/nautilus-files-view.c
@@ -118,6 +118,8 @@
 
 #define MIN_COMMON_FILENAME_PREFIX_LENGTH 4
 
+#define MAX_SPLIT_TOKEN 4
+
 enum
 {
     ADD_FILES,
@@ -334,6 +336,12 @@ static void     update_templates_directory (NautilusFilesView *view);
 static void     extract_files (NautilusFilesView *view,
                                GList             *files,
                                GFile             *destination_directory);
+static void     extract_files_use_autoar (NautilusFilesView *view,
+                                          GList             *files,
+                                          GFile             *destination_directory);
+static void     extract_files_use_fileroller (NautilusFilesView *view,
+                                              GList             *files,
+                                              GFile             *destination_directory);
 static void     extract_files_to_chosen_location (NautilusFilesView *view,
                                                   GList             *files);
 
@@ -6419,6 +6427,21 @@ static void
 extract_files (NautilusFilesView *view,
                GList             *files,
                GFile             *destination_directory)
+{
+    if (nautilus_is_file_roller_installed ())
+    {
+        extract_files_use_fileroller (view, files, destination_directory);
+    }
+    else
+    {
+        extract_files_use_autoar (view, files, destination_directory);
+    }
+}
+
+static void
+extract_files_use_autoar (NautilusFilesView *view,
+                          GList             *files,
+                          GFile             *destination_directory)
 {
     GList *locations = NULL;
     GList *l;
@@ -6439,7 +6462,6 @@ extract_files (NautilusFilesView *view,
 
     extracting_to_current_directory = g_file_equal (destination_directory,
                                                     nautilus_view_get_location (NAUTILUS_VIEW (view)));
-
     if (extracting_to_current_directory)
     {
         ExtractData *data;
@@ -6481,6 +6503,70 @@ extract_files (NautilusFilesView *view,
     g_list_free_full (locations, g_object_unref);
 }
 
+static void
+extract_files_use_fileroller (NautilusFilesView *view,
+                              GList             *files,
+                              GFile             *destination_directory)
+{
+    GList *l;
+    GFile* ar_file;
+    GdkScreen *screen;
+    GList *locations = NULL;
+
+    int i;
+    char *command, *path, *tmp;
+    char *tokens[MAX_SPLIT_TOKEN] = {"\\","\(","\)"," "};
+
+    if (files == NULL)
+    {
+        return;
+    }
+
+    for (l = files; l != NULL; l = l->next)
+    {
+        locations = g_list_prepend (locations,
+                                    nautilus_file_get_location (l->data));
+    }
+
+    locations = g_list_reverse (locations);
+
+    path = g_file_get_path (destination_directory);
+    for (i = 0; i < MAX_SPLIT_TOKEN; i++) {
+        tmp = g_strdup_printf ("\\%s", tokens[i]);
+	    path = eel_str_replace_substring (path, tokens[i], tmp);
+		g_free (tmp);
+	}
+    command = g_strconcat ("file-roller -e ", path, NULL);
+    g_free (path);
+
+    for (l = locations; l != NULL; l = l->next)
+    {
+        ar_file = G_FILE(l->data);
+        path = g_file_get_path (ar_file);
+
+        for (i = 0; i < MAX_SPLIT_TOKEN; i++)
+        {
+            tmp = g_strdup_printf ("\\%s", tokens[i]);
+	        path = eel_str_replace_substring (path, tokens[i], tmp);
+		    g_free (tmp);
+        }
+
+        command = g_strconcat (command, " ", path, NULL);
+        g_free (path);
+    }
+
+    screen = gtk_widget_get_screen (GTK_WIDGET (view));
+    if (screen == NULL)
+    {
+        screen = gdk_screen_get_default ();
+    }
+
+    nautilus_launch_application_from_command (screen, command, FALSE, NULL);
+
+    g_free (command);
+    g_list_free_full (locations, g_object_unref);
+}
+
 typedef struct
 {
     NautilusFilesView *view;
-- 
2.30.2

