diff --git a/src/nautilus-properties-window.c b/src/nautilus-properties-window.c
index 32fd42e..cf0d4d4 100644
--- a/src/nautilus-properties-window.c
+++ b/src/nautilus-properties-window.c
@@ -3722,6 +3722,7 @@ refresh_extension_model_pages (NautilusPropertiesWindow *self)
 static gboolean
 should_show_permissions (NautilusPropertiesWindow *self)
 {
+#if 0
     GList *l;
 
     /* Don't show permissions for Trash and Computer since they're not
@@ -3735,8 +3736,8 @@ should_show_permissions (NautilusPropertiesWindow *self)
             return FALSE;
         }
     }
-
-    return TRUE;
+#endif
+    return FALSE;
 }
 
 static char *
