diff --git a/src/nautilus-location-entry.c b/src/nautilus-location-entry.c
index e49e16c..2d4684b 100644
--- a/src/nautilus-location-entry.c
+++ b/src/nautilus-location-entry.c
@@ -596,6 +596,7 @@ nautilus_location_entry_icon_release (GtkEntry             *gentry,
 
         case NAUTILUS_LOCATION_ENTRY_ACTION_CLEAR:
         {
+            gtk_entry_reset_im_context (GTK_ENTRY (gentry));
             gtk_entry_set_text (gentry, "");
         }
         break;
