diff --git a/src/resources/ui/nautilus-toolbar-menu.ui b/src/resources/ui/nautilus-toolbar-menu.ui
index 384558d..d8c278f 100644
--- a/src/resources/ui/nautilus-toolbar-menu.ui
+++ b/src/resources/ui/nautilus-toolbar-menu.ui
@@ -35,6 +35,7 @@
                 <property name="position">0</property>
               </packing>
             </child>
+<!--
             <child>
               <object class="GtkButton" id="bookmark-this-location">
                 <property name="visible">True</property>
@@ -54,6 +55,7 @@
                 <property name="position">1</property>
               </packing>
             </child>
+-->
             <child>
               <object class="GtkButton" id="new-tab">
                 <property name="visible">True</property>
