diff --git a/src/nautilus-properties-window.c b/src/nautilus-properties-window.c
index c5520d2..6d3e83e 100644
--- a/src/nautilus-properties-window.c
+++ b/src/nautilus-properties-window.c
@@ -3141,6 +3141,19 @@ open_in_disks (GtkButton                *button,
     }
 }
 
+static gboolean
+check_have_gnome_disks (void)
+{
+    gchar *disks_path;
+    gboolean res;
+
+    disks_path = g_find_program_in_path ("gnome-disks");
+    res = (disks_path != NULL);
+    g_free (disks_path);
+
+    return res;
+}
+
 static void
 create_basic_page (NautilusPropertiesWindow *window)
 {
@@ -3291,14 +3304,17 @@ create_basic_page (NautilusPropertiesWindow *window)
                                                NULL);
         }
 
-       /*Translators: Here Disks mean the name of application GNOME Disks.*/
-       button = gtk_button_new_with_label (_("Open in Disks"));
-       g_signal_connect (button, "clicked", G_CALLBACK (open_in_disks), NULL);
-       gtk_container_add_with_properties (GTK_CONTAINER (grid),
-                                          button,
-                                          "width", 3,
-                                          NULL);
-       gtk_widget_show_all (GTK_WIDGET (grid));
+        if (check_have_gnome_disks ())
+        {
+            /*Translators: Here Disks mean the name of application GNOME Disks.*/
+            button = gtk_button_new_with_label (_("Open in Disks"));
+            g_signal_connect (button, "clicked", G_CALLBACK (open_in_disks), NULL);
+            gtk_container_add_with_properties (GTK_CONTAINER (grid),
+                                               button,
+                                               "width", 3,
+                                               NULL);
+        }
+        gtk_widget_show_all (GTK_WIDGET (grid));
     }
 }
 
